test/regex.test.satyg

```diff
@@ -0,0 +1,122 @@
+@require: test/test
+@require: test/expect
+
+@import: ../src/regexp
+
+let regex-test-cases = open Test in
+  open RegExp in
+  let char-a = char (Char.make `a`) in
+  let char-b = char (Char.make `b`) in
+  let char-c = char (Char.make `c`) in
+  let result-eq = Eq.equal (List.eq (Option.eq (Pair.eq Int.eq String.eq))) in
+  describe `RegExp module` [
+    % TODO: Do we need this?
+    % let () = RegExp.test (RegExp.of-string `ab.*ef`) `abcdef` |> String.of-bool |> Debug.log in
+    % let () = exec (compile (seq (many1 char-a) (many1 char-b))) `aabbb` |> String.of-bool |> Debug.log in
+    % let () = exec re-example1 `aabbb` |> String.of-bool |> Debug.log in
+    it `test "(a+)(b+)"` (fun () -> (
+      test (seq (group (many1 char-a)) (group (many1 char-b))) `aabbb`
+      |> Expect.is-true
+    ));
+    % TODO GH-142: Is this right description?
```

Comment:
I try to make descriptions from test cases, but I need more confidence with some of them. Need help from @nyuichi